crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -952,9 +952,17 @@ impl OlmMachine {
             return Ok(());
         };
 
+        // We already checked that `sender_device_keys` matches the actual sender of the
+        // message when we decrypted the message, which included doing
+        // `DeviceData::try_from` on it, so it can't fail.
```

Comment:
Can't we instead treat that the same way as we treat the `None` case up above?

And let a test ensure that we always handle this properly?

Reply:
You mean, ignore the to-device message, rather than panic?

I suppose so, though generally I take the view that "unreachable" code should cause a crash so that you get an early warning that your previous assumptions no longer hold.

Reply:
Well ok, I guess it's marginally easier to see where the assumption changed if we panic here instead in a test.

Reply:
As for a test: what did you have in mind? `receive_room_key_bundle_data` is private so can't easily be called directly from a test, as are its callers. And, for obvious reasons, we can't test this by injecting a to-device message with a mis-signed sender_device_keys.

Reply:
I just assumed that testing the happy path here would be enough to ensure that the assumption didn't change.

Reply:
If you don't have strong opinions on this one, I'd like to leave it as-is. I think otherwise we end up with more unreachable (and untestable) code.